ludwig/hyperopt/execution.py

```diff
@@ -184,121 +184,22 @@ def execute(
-            sampled_parameters = self.hyperopt_sampler.sample_batch()
-            metric_scores = []
-
-            for i, parameters in enumerate(sampled_parameters):
-                modified_config = substitute_parameters(copy.deepcopy(config), parameters)
-
-                trial_id = trials + i
-
-                model = LudwigModel(
-                    config=modified_config,
-                    backend=backend,
-                    gpus=gpus,
-                    gpu_memory_limit=gpu_memory_limit,
-                    allow_parallel_threads=allow_parallel_threads,
-                    callbacks=callbacks,
-                )
-                eval_stats, train_stats, _, _ = model.experiment(
-                    dataset=dataset,
-                    training_set=training_set,
-                    validation_set=validation_set,
-                    test_set=test_set,
-                    training_set_metadata=training_set_metadata,
-                    data_format=data_format,
-                    experiment_name=f"{experiment_name}_{trial_id}",
-                    model_name=model_name,
-                    # model_load_path=model_load_path,
-                    # model_resume_path=model_resume_path,
-                    eval_split=self.split,
-                    skip_save_training_description=skip_save_training_description,
-                    skip_save_training_statistics=skip_save_training_statistics,
-                    skip_save_model=skip_save_model,
-                    skip_save_progress=skip_save_progress,
-                    skip_save_log=skip_save_log,
-                    skip_save_processed_input=skip_save_processed_input,
-                    skip_save_unprocessed_output=skip_save_unprocessed_output,
-                    skip_save_predictions=skip_save_predictions,
-                    skip_save_eval_stats=skip_save_eval_stats,
-                    output_directory=output_directory,
-                    skip_collect_predictions=True,
-                    skip_collect_overall_stats=False,
-                    random_seed=random_seed,
-                    debug=debug,
-                )
-                metric_score = self.get_metric_score(train_stats)
-                metric_scores.append(metric_score)
-
-                trial_results.append(
-                    TrialResults(
-                        parameters=parameters,
-                        metric_score=metric_score,
-                        training_stats=train_stats,
-                        eval_stats=eval_stats,
-                    )
-                )
-            trials += len(sampled_parameters)
-
-            self.hyperopt_sampler.update_batch(zip(sampled_parameters, metric_scores))
-
-        ordered_trials = self.sort_hyperopt_results(trial_results)
-        return HyperoptResults(ordered_trials=ordered_trials)
-
-
 class RayTuneExecutor(HyperoptExecutor):
     def __init__(
         self,
         hyperopt_sampler,
```

Comment:
can we remove it?

Reply:
@w4nderlust   Not sure I know the "it" in "can we remove it?"

Reply:
Weird, if you open up Files changed it points to the right place. Anyway, I was referring to `hyperopt_sampler` in `RayTuneExecutor`. As we are already passing the search algorithm maybe we can simplify and remove the sampler alltogether?

Reply:
@w4nderlust   the answer to your question is the typical Software Engineer answer, "It depends." 😄 

The `hyperopt_sampler` is an instance of the [`RayTuneSampler` class](https://github.com/ludwig-ai/ludwig/blob/b8a180354ad965926f072df0f0221dc3a05442c5/ludwig/hyperopt/sampling.py#L67-L126).

Two methods that are needed in this class.  One is used internally in the object, the other one is invoked by the Executor.

In theory, we could migrate the methods to the `Hyperoptexecutor` superclass or its `RayTuneExecutor` subclass and then retire `RayTuneSampler` class.  This will probably take me 2 or 3 evenings to get everthing refactored and tested.

The one test I can't really do is the AutoML "getting the right answer" test.  This may mean @amholler  will need to redo the testing that is occuring right now.

Or we can leave it as is now and do the refactoring in a later release as part of code clean-up.  As best I can tell, this change is purely internal and does not affect any external facing APIs.  I don't think this will be a breaking change.

Let me know how you'd like me to proceed.

Reply:
Got it, makes sense. The way I think about it is that logically what the sampler was doing before was implementing a search algorithm, so now having the search algorithm itself, it feels like logically there shouldn't be a need for it, although I would need to look more closely at the entire code after refactoring to understand the implications of removing the sampler.

Anyway, because this would be a change that does not affect the interface, but an internal reorganization, maybe we can land this PR first, which unlocks Anne and others to adapt to the changes in the interface, and then consider doing a followup pr to remove cleanup the sampler. What do you think?

Reply:
@w4nderlust   I think it is OK to leave it as is for now.  Maybe get a few other eyes on the sampler part and get feedback from others on the refactored hyperopt.  We can decide later on how much more code refactoring is needed.    So I'll mark this as resolved for now.  I'll focus on finishing up the doc updates in by this weekend.  this should put the PR in  a good place for the final merge of code and docs.